src/query/mlt/query.rs

```diff
@@ -0,0 +1,284 @@
+use super::MoreLikeThis;
+
+use crate::{
+    query::{Query, Weight},
+    schema::{Field, FieldValue},
+    DocAddress, Result, Searcher, TantivyError,
+};
+
+/// A query that matches all of the documents similar to a document
+/// or a set of field values provided.
+///
+/// # Examples
+///
+/// ```
+/// use tantivy::DocAddress;
+/// use tantivy::query::MoreLikeThisQuery;
+///
+/// let query = MoreLikeThisQuery::builder()
+///     .with_min_doc_frequency(1)
+///     .with_max_doc_frequency(10)
+///     .with_min_term_frequency(1)
+///     .with_min_word_length(2)
+///     .with_max_word_length(5)
+///     .with_boost_factor(1.0)
+///     .with_stop_words(vec!["for".to_string()])
+///     .with_document(DocAddress::new(2, 1));
+///
+/// ```
+#[derive(Debug, Clone)]
+pub struct MoreLikeThisQuery {
+    mlt: MoreLikeThis,
+    doc_address: Option<DocAddress>,
```

Comment:
we could use a small private enum to replace the two option and the error handling.